kqueen/engines/openstack_kubespray.py

```diff
@@ -334,10 +345,15 @@ def resize(self, node_count):
         node_count = int(node_count)
         master_count = len(self.cluster.metadata["resources"]["masters"])
         new_slave_count = node_count - master_count
-        if new_slave_count < 0:
-            return False, "Node count should be at least %s" % master_count
+        if new_slave_count <= 0:
+            return False, "Node count should be at least %s" % (master_count + 1)
         current_slave_count = len(self.cluster.metadata["resources"]["slaves"])
         delta = new_slave_count - current_slave_count
+
+        if delta == 0:
+            return False, "Cluster already has %s nodes" % node_count
+        self.cluster.state = config.CLUSTER_UPDATING_STATE
+        self.cluster.save()
```

Comment:
raw 361, maybe just else?

Reply:
elif redundant)

Reply:
Didn't get you, I didnt change 361 raw. Are you suggesting to to rename elif to else?

Reply:
yep, i undrestand, its minor ) but if u have time )